internal/field_tree_printer/field_tree_printer.go

```diff
@@ -3,48 +3,53 @@ package field_tree_printer
 import (
 	"strings"
 
-	"github.com/codecrafters-io/kafka-tester/internal/field_decoder"
 	"github.com/codecrafters-io/kafka-tester/internal/field_path"
+	"github.com/codecrafters-io/kafka-tester/protocol/value"
 	"github.com/codecrafters-io/tester-utils/logger"
 )
 
+type Field interface {
+	GetPath() field_path.FieldPath
+	GetValue() value.KafkaProtocolValue
+}
+
 type FieldTreePrinter struct {
-	DecodedFields []field_decoder.DecodedField
-	Logger        *logger.Logger
+	Fields []Field
+	Logger *logger.Logger
 
 	currentIndentationLevel int
 	lastPrintedFieldPath    field_path.FieldPath
 }
 
-func (p FieldTreePrinter) PrintForErrorLogs(errorPath field_path.FieldPath) {
+func (p FieldTreePrinter) PrintForErrorLogs(errorPath field_path.FieldPath, errorMessage string) {
 	p.currentIndentationLevel = 0
 	p.lastPrintedFieldPath = field_path.NewFieldPath("")
 
-	for _, decodedField := range p.DecodedFields {
-		p.printNodesLeadingTo(decodedField.Path, p.Logger.Infof)
+	for _, field := range p.Fields {
+		p.printNodesLeadingTo(field.GetPath(), p.Logger.Infof)
 
-		if errorPath.Is(decodedField.Path) {
-			p.Logger.Infof("%s❌ %s (%s)", p.buildIndentPrefix(), decodedField.Path.LastSegment(), decodedField.Value.String())
+		if errorPath.Is(field.GetPath()) {
+			p.Logger.Infof("%s❌ %s (%s)", p.buildIndentPrefix(), field.GetPath().LastSegment(), field.GetValue().String())
```

Comment:
If "errorMessage" is passed in, I'd expect it to be used here too. It isn't intuitive that we pass in error message but it isn't used when printing the wrong value. Earlier this was fine since the method was in the context of decoded values and could intelligently decide when to say "decode error" vs. value error